app/models/collection.rb

```diff
@@ -1,5 +1,5 @@
 class Collection < ApplicationRecord
-  has_many :articles
+  has_many :articles, dependent: :nullify
```

Comment:
Destroying a collection simply means removing the articles from the collection, not destroying the articles